service/server.py

```diff
@@ -12,41 +12,51 @@ def get_register_title(title_ref):
 
 @app.route('/titles/<title_ref>', methods=['GET'])
 def display_title(title_ref):
+    #TODO Refactoring is needed, as well commenting on the code.
     api_response = get_register_title(title_ref)
     if api_response:
         title_api = api_response.json()
-        entries = title_api['data']['entries']
-        property_description = {}
-        for entry in entries:
-            #TODO: story US25
-            #if entry['role_code'] == 'RPRO':
-            #    infills = entry['infills']
-            #        for infill in infills:
-                    #Get the infill of type address
-            #        proprietors = infill['proprietors']
-            #        for proprietor in prosprietors:
-            #            proprietor_names += [proprietor['name']['forename'] + ' ' + proprietor['name']['surname']]
-            if entry['role_code'] == 'RDES':
-                infills = entry['infills']
-                for infill in infills:
-                    address_part = infill['address']
-                    if address_part:
-                        first_line_address = address_part['house_no'] + ' ' + address_part['street_name']
-                        property_description = {
-                            'first_line': first_line_address,
-                            'town': address_part['town'],
-                            'postcode': address_part['postcode']
-                        }
+        entry_groups = title_api['data']['groups']
+        #This is to get the proprietor entry that is in the ownership category
+        proprietor_names = []
+        for entry_group in entry_groups:
```

Comment:
It would be best to refactor these structures of fors and ifs by using functional approach (mapping and filtering with lambdas). Can also extract these two pieces into separate functions. I realise this code was much like that before.